src/reason-parser/reason_pprint_ast.ml

```diff
@@ -6596,17 +6583,47 @@ let printer = object(self:'self)
           ~comments:self#comments
           s
         in
-        makeList
-          ~break:IfNeed
-          ~inline:(true, false)
-          ~wrap:("{", "}")
-          ~postSpace:true
-          ~sep:(SepFinal (";", ";"))
-          items
-      | Pmty_extension (s, e) -> self#payload "%" s e
-      | _ -> makeList ~break:IfNeed ~wrap:("(", ")") [self#module_type x]
+        let someSigHasAttrs =
+          List.exists
```

Comment:
Instead of inspecting the children here, you can use `Always_rec` deeper in the tree.
This recursively breaks nodes from the child up to the root of the tree.